server/src/main/java/org/opensearch/index/engine/exec/merge/MergeHandler.java

```diff
@@ -83,22 +95,22 @@ public synchronized void registerMerge(OneMerge merge) {
                 }
             }
         } catch (Exception e) {
+            logger.warn(() -> new ParameterizedMessage("Failed to acquire snapshots", e));
             throw new RuntimeException(e);
         }
         mergingSegments.add(merge);
```

Comment:
We can add debug log when registering merge and also log mergingSegments as well.